### DISCUSSION TITLE
`libpack` vs `packit`

### DISCUSSION THREAD

**Discussion by braunsonm** [paketo-buildpacks/feedback#256] [Builder Team]
I'm looking at contributing to Paketo and I see some buildpacks use `libpack` while others use `packit` what is the difference between these two libraries for creating CNB's? And when should you use each?

Thanks!

**Comment by dmikusa** [comment]
The difference is in the API that they provide. Both implement the buildpack specification, so at the core, they are the same but the way you do certain things in the libraries differs. It's hard to compare because I don't use both libraries, so while I can tell you a lot about libpak I can't tell you much about packit. You can look at the Go doc for the libraries if you want to see more about the APIs, or just compare the patterns in a buildpack written with each library.

Most people pick one or the other, and typically you pick based on the buildpacks you want to contribute to. Each buildpack is written using one of the libraries and you wouldn't want to mix the two. All of the Java, most of the Utilities, and all of the APM buildpacks use libpak, and the others use packit. Hope that helps!
